Client/Helpers/TabEventHandler.swift

```diff
@@ -100,22 +91,11 @@ enum TabEvent {
     case didDeriveMetadata(DerivedMetadata)
 
     var label: TabEventLabel {
-        switch self {
-        case .didChangeURL:
-            return .didChangeURL
-        case .didLoadPageMetadata:
-            return .didLoadPageMetadata
-        case .didLoadFavicon:
-            return .didLoadFavicon
-        case .didGainFocus:
-            return .didGainFocus
-        case .didLoseFocus:
-            return .didLoseFocus
-        case .didClose:
-            return .didClose
-        case .didDeriveMetadata:
-            return .didDeriveMetadata
+        let str = "\(self)".components(separatedBy: "(")[0] // Will grab just the name from 'didChangeURL(...)'
```

Comment:
I'm trying to think if there's any way we can put an assertion or something to prevent someone down the road from accidentally duping a method name. Maybe just put a comment above to clearly state that method names in this protocol need to be unique for this to work.

Reply:
Didn't think of that! Good point, I'll add it.